src/importer.ts

```diff
@@ -30,35 +30,45 @@ export class Importer {
 
     public importAllEvents(): void {
         let runCount = 0;
+        let amqpChannel;
         this.amqpConnectionProvider
             .flatMap(connection => connection.createChannel())
             .flatMap(channel => channel.assertQueue(this.config.amqp.queues.event, { durable: true }))
             .flatMap(reply => {
+                amqpChannel = reply.channel;
                 reply.channel.prefetch(this.config.maxBatchSize);
                 return reply.channel.consume(this.config.amqp.queues.event, { noAck: false });
             })
             .bufferWithTimeOrCount(this.config.maxBatchWaitTimeMs, this.config.maxBatchSize)
-            .subscribe(this.runBatchThenAck.bind(this, runCount));
+            .subscribe(amqpMessages => this.runBatchThenAck(runCount, amqpChannel, amqpMessages));
     }
 
-    private runBatchThenAck(runCount: number, events) {
-        if (events.length === 0) {
+    private runBatchThenAck(runCount: number, amqpChannel, amqpMessages) {
+        if (amqpMessages.length === 0) {
             logger.info("Empty buffer, skipping it");
             return;
         }
 
         runCount++;
-        logger.info("Cycle %d has %d events", runCount, events.length);
-        this.runBatchOnPapi(events)
-            .then(batchResult => {
-                logger.info("Batch %d is done: ", runCount, batchResult.message);
-                batchResult.errors.forEach(e => batchErrorLogger.error("Batch #%d", runCount, e));
-            })
-            .then(() => {
-                setTimeout(() => events.forEach(e => e.ack()), this.config.delayBetweenBatchMs);
-            });
+        logger.info("Batch %d has %d events", runCount, amqpMessages.length);
+        this.runBatchOnPapi(amqpMessages).then(batchResult => {
+            logger.info("Batch %d is done: ", runCount, batchResult.message);
+            batchResult.errors.forEach(e => batchErrorLogger.error("Batch #%d", runCount, e));
+            setTimeout(() => this.ackOrRequeueAllIfAnyError(batchResult, amqpChannel, amqpMessages), this.config.delayBetweenBatchMs);
+        });
     }
 
+    private ackOrRequeueAllIfAnyError(batchResult, amqpChannel, amqpMessages) {
```

Comment:
I suggest to not use RabbitMQ-related types/objects in message consumer implementation, and stick with business logic only (consumers/publishers should not always know about the underlying message broker IMO).
For instance, an abstraction can be passed that would handle delivering an error message to the configured queue, through a publish(Error) method.